sources/api/apiserver/src/server/ephemeral_storage.rs

```diff
@@ -114,10 +114,13 @@ pub fn bind(variant: &str, dirs: Vec<String>) -> Result<()> {
 
     let mount_point = format!("/mnt/{}", EPHEMERAL_MNT);
     let mount_point = Path::new(&mount_point);
-    let allowed_dirs = allowed_bind_dirs(variant);
+    let (allowed_exact, allowed_prefixes) = allowed_bind_dirs(variant);
```

Comment:
Two concerns:
1. We should disallow paths starting with `/mnt/.ephemeral` (the value in `mount_point`) since that directory and its subdirectories are used for keeping state and siloing content of the different mount points.
2. We should reject any paths that contain `..` as that could traverse outside `/mnt` or over-mount other paths within it.

Reply:
@bcressey thanks for the feedback, I have added checks for presence of .. and /mnt/.ephemeral in the next commit.